core/services/src/main/java/org/exoplatform/services/cms/clouddrives/DisconnectCloudDriveServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+package org.exoplatform.services.cms.clouddrives;
+
+import org.exoplatform.documents.service.DocumentFileService;
+import org.exoplatform.services.cms.clouddrives.jcr.NodeFinder;
+import org.exoplatform.services.cms.documents.FavoriteService;
+import org.exoplatform.services.jcr.RepositoryService;
+import org.exoplatform.services.jcr.ext.app.SessionProviderService;
+import org.exoplatform.services.jcr.ext.common.SessionProvider;
+import org.exoplatform.services.jcr.impl.core.NodeImpl;
+import org.exoplatform.services.security.ConversationState;
+import org.exoplatform.services.security.Identity;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+
+import javax.jcr.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class DisconnectCloudDriveServiceImpl implements DisconnectCloudDriveService {
+
+  private DocumentFileService    documentFileService;
+
+  private SessionProviderService sessionProviders;
+
+  private NodeFinder             finder;
+
+  private RepositoryService      jcrService;
+
+  private IdentityManager        identityManager;
+
+  private FavoriteService        favoriteService;
+
+  public DisconnectCloudDriveServiceImpl(DocumentFileService documentFileService,
+                                         SessionProviderService sessionProviders,
+                                         NodeFinder finder,
+                                         RepositoryService jcrService,
+                                         IdentityManager identityManager,
+                                         FavoriteService favoriteService) {
+    this.documentFileService = documentFileService;
+    this.sessionProviders = sessionProviders;
+    this.finder = finder;
+    this.jcrService = jcrService;
+    this.identityManager = identityManager;
+    this.favoriteService = favoriteService;
+  }
+
+  @Override
+  public void disconnectCloudDrive(String workspace, String path) {
+    long userIdentityId = getCurrentUserIdentityId();
+    String userName = getCurrentUserIdentity().getUserId();
+    SessionProvider sp = sessionProviders.getSessionProvider(null);
+    Session userSession = null;
+    Item item = null;
+    List<Node> nodes = new ArrayList<Node>();
+    long delay = 6;
+    try {
+      userSession = sp.getSession(workspace, jcrService.getCurrentRepository());
+      item = finder.findItem(userSession, path);
+      Node userNode = (Node) item;
+      Node childNode = null;
+      NodeIterator nodeIterator = userNode.getNodes();
+      while (nodeIterator.hasNext()) {
+        childNode = nodeIterator.nextNode();
+        if (childNode.hasProperty("ecd:connected")) {
+          nodes.add(childNode);
+        }
+      }
```

Comment:
you don't need to get the child nodes IMHo, you can delete directly the needed node